packages/node_modules/nav-frontend-stegindikator/src/Stepper.tsx

```diff
@@ -0,0 +1,240 @@
+    const [compact, setCompact] = useState(false);
+
+    const list = useRef<HTMLOListElement>(null);
+
+    const getDefaultActiveStegIndex = () => {
+        let index;
+        if (children) {
+            React.Children.forEach(children, (child, i) => {
+                if (React.isValidElement(child)) {
+                    const clone = React.cloneElement(child as React.ReactElement<any>);
+                    if (clone.props['aktiv']) {
+                        index = i;
+                    }
+                }
+            });
+        } else {
+            index = steg!.findIndex((steg) => !!steg.aktiv);
+        }
+        return (index !== -1) ? index : 0 ;
+    };
+
+    const getNumSteg = () => {
+        if (children) {
+            return React.Children.toArray(children).filter((child) => React.isValidElement(child)).length;
+        }
+        return steg!.length;
+    };
+
+    const getDimensions = () => {
+        const numSteg = getNumSteg();
+        const remSize = parseFloat(String(getComputedStyle(document.documentElement).fontSize));
+        const margin = remSize * 1.25;
+        const marginTotal = (margin * numSteg) - margin;
+        const visLabelWidth = ((remSize * 10) * numSteg) + marginTotal;
+        const normalWidth = ((remSize * 2) * numSteg) + marginTotal;
+
+        return {
+            visLabelWidth,
+            normalWidth,
+            container: list.current!.getBoundingClientRect().width
+        };
+    };
+
+    const adjustSize = () => {
+        if (!list.current) return;
+
+        setShowLabel(canShowLabel());
+        setCompact(!canBeNormal());
+    };
+
+    const canShowLabel = () => {
+        const dim = getDimensions();
+        return dim.container >= dim.visLabelWidth && visLabel;
+    };
+
+    const canBeNormal = () => {
+        const dim = getDimensions();
+        return dim.container >= dim.normalWidth && !kompakt;
+    };
+
+    const initialSteg = () => {
+        let initialAktivtSteg;
+        if (aktivtSteg !== undefined) {
+            initialAktivtSteg = aktivtSteg;
+        } else {
+            initialAktivtSteg = getDefaultActiveStegIndex();
+        }
+        return initialAktivtSteg;
+    };
+
+    // Kjører etter komponent er byggd, erstatter "constructor" siden den kjører bare en gang
+    useEffect(() => {
+        setActiveStep(initialSteg());
+        setCompact(kompakt);
+        setShowLabel(visLabel);
+
+        if (autoResponsiv) {
+            adjustSize();
+        }
+    },        []);
+
+    // ComponentWillRecieveProps alternativ, men kjører bare for spesifikke props
+    useEffect(() => {
+        if (autoResponsiv) {
+            const label = (canShowLabel()) ? visLabel : false;
+            setShowLabel(label);
+            setCompact(kompakt || !canBeNormal());
+        } else {
+            setShowLabel(visLabel);
+            setCompact(kompakt);
+        }
+    },        [autoResponsiv, kompakt, visLabel]);
```

Comment:
eslint reglene for hooks vil vel klage her? E.g missing dependency `canShowLabel` etc
Mulig de reglene ikke er satt opp her, men vi bør nok sette de opp om vi skal benytte hooks.